test/utils/shoots/update/update.go

```diff
@@ -44,9 +64,36 @@ func RunTest(
 	newWorkerPoolKubernetesVersion *string,
 ) {
 	By("creating shoot client")
+	var (
+		job *batchv1.Job
+		err error
+	)
+
 	shootClient, err := access.CreateShootClientFromAdminKubeconfig(ctx, f.GardenClient, f.Shoot)
 	Expect(err).NotTo(HaveOccurred())
 
+	if gardencorev1beta1helper.IsHAControlPlaneConfigured(f.Shoot) {
+		f.Seed, f.SeedClient, err = f.GetSeed(ctx, *f.Shoot.Spec.SeedName)
+		Expect(err).NotTo(HaveOccurred())
+		shootSeedNamespace := f.Shoot.Status.TechnicalID
+		By("deploying zero-downtime validator job")
+		job, err = highavailability.DeployZeroDownTimeValidatorJob(ctx,
+			f.SeedClient.Client(), "update", shootSeedNamespace, getKubeAPIServerAuthToken(ctx, f.SeedClient, shootSeedNamespace))
+		Expect(err).NotTo(HaveOccurred())
+
+		waitForJobToBeReady(ctx, f.SeedClient.Client(), job)
+		defer func() {
+			ExpectWithOffset(1,
+				client.IgnoreNotFound(
+					f.SeedClient.Client().Delete(ctx,
+						job,
+						client.PropagationPolicy(metav1.DeletePropagationForeground),
+					),
+				),
+			).To(Succeed())
+		}()
```

Comment:
Use `DeferCleanup` instead?

Reply:
DeferCleanup will not help in this case, because `DeferCleanup` is called at the end of `It` block and shoot is deleted as part of `It` block.

Reply:
OK, then maybe move this deletion down to the end of the function to get rid of this `defer` statement here